Does not build with g++-8

both 0.2.0 and master fail to build. It also failed to build on debian sid with similar llvm and clang.

```
Package: libxml2-dev
Version: 2.9.4+dfsg1-6.1ubuntu1

Package: libclang-6.0-dev
Version: 1:6.0.1-2ubuntu1

```
```
[20/233] Building CXX object CMakeFiles/embedded_lld_elf.dir/deps/lld/ELF/AArch64ErrataFix.cpp.o
FAILED: CMakeFiles/embedded_lld_elf.dir/deps/lld/ELF/AArch64ErrataFix.cpp.o 
/usr/lib/distcc/c++   -Ideps/lld/include -I/usr/lib/llvm-6.0/include -Ideps/lld/ELF -Ideps/lld-prebuilt/ELF -Ideps/lld-prebuilt -Ideps/SoftFloat-3e/source/include -I. -Isrc -g -Werror -Wall   -std=c++11 -fno-exceptions -fno-rtti -Wno-comment -MD -MT CMakeFiles/embedded_lld_elf.dir/deps/lld/ELF/AArch64ErrataFix.cpp.o -MF CMakeFiles/embedded_lld_elf.dir/deps/lld/ELF/AArch64ErrataFix.cpp.o.d -o CMakeFiles/embedded_lld_elf.dir/deps/lld/ELF/AArch64ErrataFix.cpp.o -c deps/lld/ELF/AArch64ErrataFix.cpp
In file included from /usr/lib/llvm-6.0/include/llvm/ADT/STLExtras.h:21,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Host.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Hashing.h:49,
                 from deps/lld/include/lld/Common/LLVM.h:20,
                 from deps/lld/ELF/AArch64ErrataFix.h:13,
                 from deps/lld/ELF/AArch64ErrataFix.cpp:32:
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h: In instantiation of ‘void llvm::SmallVectorTemplateBase<T, true>::push_back(const T&) [with T = std::pair<void*, long unsigned int>]’:
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:247:33:   required from ‘void* llvm::BumpPtrAllocatorImpl<AllocatorT, SlabSize, SizeThreshold>::Allocate(size_t, size_t) [with AllocatorT = llvm::MallocAllocator; long unsigned int SlabSize = 4096; long unsigned int SizeThreshold = 4096; size_t = long unsigned int]’
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:57:67:   required from ‘void* llvm::AllocatorBase<DerivedT>::Allocate(size_t, size_t) [with DerivedT = llvm::BumpPtrAllocatorImpl<>; size_t = long unsigned int]’
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:79:12:   required from ‘T* llvm::AllocatorBase<DerivedT>::Allocate(size_t) [with T = lld::elf::Patch843419Section; DerivedT = llvm::BumpPtrAllocatorImpl<>; size_t = long unsigned int]’
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:434:65:   required from ‘T* llvm::SpecificBumpPtrAllocator<T>::Allocate(size_t) [with T = lld::elf::Patch843419Section; size_t = long unsigned int]’
deps/lld/include/lld/Common/Memory.h:55:10:   required from ‘T* lld::make(U&& ...) [with T = lld::elf::Patch843419Section; U = {lld::elf::InputSection*&, long unsigned int&}]’
deps/lld/ELF/AArch64ErrataFix.cpp:562:56:   required from here
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h:312:11: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct std::pair<void*, long unsigned int>’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
     memcpy(this->end(), &Elt, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/8/utility:70,
                 from /usr/lib/llvm-6.0/include/llvm/Support/type_traits.h:19,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Optional.h:22,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/STLExtras.h:20,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Host.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Hashing.h:49,
                 from deps/lld/include/lld/Common/LLVM.h:20,
                 from deps/lld/ELF/AArch64ErrataFix.h:13,
                 from deps/lld/ELF/AArch64ErrataFix.cpp:32:
/usr/include/c++/8/bits/stl_pair.h:198:12: note: ‘struct std::pair<void*, long unsigned int>’ declared here
     struct pair
            ^~~~
cc1plus: all warnings being treated as errors
distcc[11068] ERROR: compile deps/lld/ELF/AArch64ErrataFix.cpp on localhost failed
[25/233] Building CXX object CMakeFiles/zig.dir/src/translate_c.cpp.o
FAILED: CMakeFiles/zig.dir/src/translate_c.cpp.o 
/usr/lib/distcc/c++   -Ideps/lld/include -I/usr/lib/llvm-6.0/include -Ideps/SoftFloat-3e/source/include -I. -Isrc -Ideps/SoftFloat-3e-prebuilt -Ideps/SoftFloat-3e/source/8086 -g -Werror -Wall   -std=c++11 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_GNU_SOURCE -fno-exceptions -fno-rtti  -Werror=strict-prototypes -Werror=old-style-definition -Werror=type-limits -Wno-missing-braces -MD -MT CMakeFiles/zig.dir/src/translate_c.cpp.o -MF CMakeFiles/zig.dir/src/translate_c.cpp.o.d -o CMakeFiles/zig.dir/src/translate_c.cpp.o -c src/translate_c.cpp
In file included from /usr/lib/llvm-6.0/include/llvm/ADT/STLExtras.h:21,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Host.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Hashing.h:49,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/ArrayRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/APFloat.h:21,
                 from /usr/lib/llvm-6.0/include/clang/AST/APValue.h:18,
                 from /usr/lib/llvm-6.0/include/clang/AST/Decl.h:17,
                 from /usr/lib/llvm-6.0/include/clang/AST/ASTTypeTraits.h:20,
                 from /usr/lib/llvm-6.0/include/clang/AST/ASTContext.h:18,
                 from /usr/lib/llvm-6.0/include/clang/Frontend/ASTUnit.h:18,
                 from src/translate_c.cpp:18:
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h: In instantiation of ‘void llvm::SmallVectorTemplateBase<T, true>::push_back(const T&) [with T = std::pair<void*, long unsigned int>]’:
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:247:33:   required from ‘void* llvm::BumpPtrAllocatorImpl<AllocatorT, SlabSize, SizeThreshold>::Allocate(size_t, size_t) [with AllocatorT = llvm::MallocAllocator; long unsigned int SlabSize = 4096; long unsigned int SizeThreshold = 4096; size_t = long unsigned int]’
/usr/lib/llvm-6.0/include/clang/AST/ASTContext.h:651:42:   required from here
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h:312:11: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct std::pair<void*, long unsigned int>’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
     memcpy(this->end(), &Elt, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/8/utility:70,
                 from /usr/lib/llvm-6.0/include/llvm/Support/type_traits.h:19,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Casting.h:19,
                 from /usr/lib/llvm-6.0/include/clang/Basic/LLVM.h:22,
                 from /usr/lib/llvm-6.0/include/clang/AST/APValue.h:17,
                 from /usr/lib/llvm-6.0/include/clang/AST/Decl.h:17,
                 from /usr/lib/llvm-6.0/include/clang/AST/ASTTypeTraits.h:20,
                 from /usr/lib/llvm-6.0/include/clang/AST/ASTContext.h:18,
                 from /usr/lib/llvm-6.0/include/clang/Frontend/ASTUnit.h:18,
                 from src/translate_c.cpp:18:
/usr/include/c++/8/bits/stl_pair.h:198:12: note: ‘struct std::pair<void*, long unsigned int>’ declared here
     struct pair
            ^~~~
cc1plus: all warnings being treated as errors
distcc[11059] ERROR: compile src/translate_c.cpp on localhost failed
[29/233] Building CXX object CMakeFiles/zig_cpp.dir/src/zig_llvm.cpp.o
FAILED: CMakeFiles/zig_cpp.dir/src/zig_llvm.cpp.o 
/usr/lib/distcc/c++   -Ideps/lld/include -I/usr/lib/llvm-6.0/include -Ideps/SoftFloat-3e/source/include -I. -Isrc -g -Werror -Wall   -std=c++11 -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -D_GNU_SOURCE -fno-exceptions -fno-rtti  -Werror=strict-prototypes -Werror=old-style-definition -Werror=type-limits -Wno-missing-braces -MD -MT CMakeFiles/zig_cpp.dir/src/zig_llvm.cpp.o -MF CMakeFiles/zig_cpp.dir/src/zig_llvm.cpp.o.d -o CMakeFiles/zig_cpp.dir/src/zig_llvm.cpp.o -c src/zig_llvm.cpp
distcc[11072] ERROR: compile src/zig_llvm.cpp on 10.66.3.7/6,lzo failed
distcc[11072] (dcc_build_somewhere) Warning: remote compilation of 'src/zig_llvm.cpp' failed, retrying locally
distcc[11072] Warning: failed to distribute src/zig_llvm.cpp to 10.66.3.7/6,lzo, running locally instead
In file included from /usr/lib/llvm-6.0/include/llvm/ADT/STLExtras.h:21,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Host.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Hashing.h:49,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/ArrayRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/DenseMapInfo.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/DenseMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Analysis/TargetLibraryInfo.h:13,
                 from src/zig_llvm.cpp:18:
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h: In instantiation of ‘void llvm::SmallVectorTemplateBase<T, true>::push_back(const T&) [with T = std::pair<void*, long unsigned int>]’:
/usr/lib/llvm-6.0/include/llvm/Support/Allocator.h:247:33:   required from ‘void* llvm::BumpPtrAllocatorImpl<AllocatorT, SlabSize, SizeThreshold>::Allocate(size_t, size_t) [with AllocatorT = llvm::MallocAllocator; long unsigned int SlabSize = 4096; long unsigned int SizeThreshold = 4096; size_t = long unsigned int]’
/usr/lib/llvm-6.0/include/llvm/Support/YAMLParser.h:130:42:   required from here
/usr/lib/llvm-6.0/include/llvm/ADT/SmallVector.h:312:11: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘struct std::pair<void*, long unsigned int>’ with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
     memcpy(this->end(), &Elt, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/8/utility:70,
                 from /usr/lib/llvm-6.0/include/llvm/Support/type_traits.h:19,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Optional.h:22,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/STLExtras.h:20,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/StringMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Support/Host.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/Hashing.h:49,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/ArrayRef.h:13,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/DenseMapInfo.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/ADT/DenseMap.h:17,
                 from /usr/lib/llvm-6.0/include/llvm/Analysis/TargetLibraryInfo.h:13,
                 from src/zig_llvm.cpp:18:
/usr/include/c++/8/bits/stl_pair.h:198:12: note: ‘struct std::pair<void*, long unsigned int>’ declared here
     struct pair
            ^~~~
cc1plus: all warnings being treated as errors
distcc[11072] ERROR: compile src/zig_llvm.cpp on localhost failed
ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -774,7 +774,7 @@ if(MSVC)
 elseif(MINGW)
   set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Werror -Wno-error=format= -Wno-error=format -Wno-error=format-extra-args")
 else()
-  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Werror -Wall")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Werror -Wall -Wno-unknown-warning-option -Wno-error=class-memaccess")
 endif()
 
 if(MSVC)

